lib/money/splitter.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+class Money
+  class Splitter
+    include Enumerable
+
+    attr_reader :split
+    protected attr_writer :split
+
+    def initialize(money, num)
+      num = Integer(num)
+      raise ArgumentError, "need at least one party" if num < 1
+      subunits = money.subunits
+      low = Money.from_subunits(subunits / num, money.currency)
+      high = Money.from_subunits(low.subunits + 1, money.currency)
+
+      num_high = subunits % num
+
+      @split = {}
+      @split[high] = num_high if num_high > 0
+      @split[low] = num - num_high
```

Comment:
I'd recommend moving this out of the initializer. We only need to compute this once it's used. It could also probably prevent us from having to `dup` above, by creating a new hash (although with the freeze, I'm not sure if dup is needed at all)

Reply:
> There will likely be cases where this object is created but never used

I doubt so, but ok.